### ISSUE TITLE
How to declaratively apply Converters to a Type or Member recursively

### ISSUE THREAD

**Issue by bartelink** [JamesNK/Newtonsoft.Json#1827]
I have an expiring bounty on an SO question and thought I might log it here. (I'm good for the bounty even after it has expired ;) ... but the fake internet points are about to go up in smoke and someone might be watching that knows just how to accomlush this!)

https://stackoverflow.com/questions/51077670/getting-converters-specified-via-attributes-to-be-equivalent-to-global-ones-in-j

Will expand on the question here with a repro fiddle, but don't have time today, and the clock is ticking!

**Comment by bartelink** [comment]
@JamesNK I've edited the question to look more like the fully-formed question it should have been in the first instance; my apologies for the presumptiveness of dumping a link to a SO question as I did, no matter how well intentioned I reasoned I was being.

If it would help, I can clarify and/or set up a fiddle for you to play with to demonstrate the problem better. By the same token, I'm happy with a basic confirmation that I'm correct that this does not exist and there is no easy way to achieve it.

**Comment by bartelink** [comment]
Ping @JamesNK Any chance of a response? This is a very relevant question for large codebases IMO and I'd love to know if there is a way to achieve this result. If you wish I can repost the above as a new question with the amended text above

**Comment by bartelink** [comment]
@JamesNK I'm still interested in the answer to this question and would prefer not to have to re-log it (or annoyingly be pinging you like this)

TL;DR is there a way to resolve this; is it possible to write a converter ? an I misunderstanding how some other technique can achieve this? is it conceivable that this'd become a first class feature if there is no present approach to this? 

It's a big win for me if I can get out of the game of addressing conversions
- by playing with fire by having potentially conflicting global converters
- risking introducing unintended side-effects from an over-eager converter

And instead be able to tag stuff explicitly, which
- constrains the scope of the conversion's effect
- serves to document where a conversion is in effect

Context: I'm using F# and am we're attempting to use only [these converters](https://github.com/jet/Jet.JsonNet.Converters) - there's a school of thought that using explicit serialization (which in general I abhor) at least pins a behaviour (yes, sometimes for the worse). When you've got hundreds/thousands of records in a codebase, being able to avoid throwing in another converter and hope for the best is a big risk (to be clear: in this case I wouldnt even be able to completely solve the problem with a global converter in the first instance)

**Comment by bartelink** [comment]
@JamesNK bump ?

**Comment by bartelink** [comment]
@JamesNK Trying to avoid pasting this all into a new issue to get attention; I'll write a self-answered SO post covering this scenario iff we can work out a way to address it in some manner
